src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySinkConfig.java

```diff
@@ -104,12 +105,24 @@ public boolean isAllowSchemaRelaxation() {
   public void validate(FailureCollector collector) {
     super.validate(collector);
     String bucket = getBucket();
-    if (!containsMacro(NAME_BUCKET) && bucket != null) {
-      // Basic validation for allowed characters as per https://cloud.google.com/storage/docs/naming
-      Pattern p = Pattern.compile("[a-z0-9._-]+");
-      if (!p.matcher(bucket).matches()) {
-        collector.addFailure("Bucket must only contain lowercase characters, numbers, '.', '_', and '-'", null)
-          .withConfigProperty(NAME_BUCKET);
+    // Allowed character validation for bucket name as per https://cloud.google.com/storage/docs/naming
+    String errorMessage = "Bucket name can contain following characters: lowercase letters, numbers, " +
+      "'.', '_', and '-'.";
+    match(bucket, "[a-z0-9._-]+", collector, errorMessage, NAME_BUCKET);
+    // Allowed character validation for dataset name as per https://cloud.google.com/bigquery/docs/datasets
+    errorMessage = "Dataset name can contain following characters: letters (lower or uppercase), numbers or '_'.";
+    match(dataset, "[A-Za-z0-9_]+", collector, errorMessage, NAME_DATASET);
+  }
+
+  /**
+   * Matches text with provided pattern. If the text does not match the pattern, add a new failure to failure collector.
+   *
+   */
+  protected void match(String text, String pattern, FailureCollector collector, String errorMessage, String property) {
```

Comment:
it's a little better to pass in a Pattern instead of a String and have constants for the Patterns instead of re-using hardcoded strings.

This method can also be moved to some common place instead of being implemented in 2 plugins.